### ISSUE TITLE
Wrong fields inferencing

### ISSUE THREAD

**Issue by aborruso** [nushell/nushell#7622]
### Describe the bug

Hi,
daily I work with data where some columns have only numerical values in the cells, but which are actually string fields.

A typical case is that of administrative codes, of regions, provinces and cities (in the table below `codice_regione`, `codice_provincia` and `codice_comune` fields). In Italy, the city of Losine has the code `0870`, and obviously it's a string, and it must remain a string, otherwise I would lose the chance to JOIN this data with other data.

If I run

```
let df = open-df --infer-schema 200000 input.csv
```

the `codice_regione`, `codice_provincia` and `codice_comune` fields are no longer strings, they are all numbers and the cell values have changed (in example the code of Losine becomes `870`). 
Four fields are modified and no longer usable.

I know, octal numbers exist, but in 99% of cases (I'm talking about my experience), if I have fields with cells starting with zero they are not octal numbers, but string codes.
And I know I can force all to strings, but then I must correct all numeric fields.

If it were possible I would add a subrule for all fields mapped as numbers: if there are cells that start with a zero,  not followed by a `,` or `.`, set that inferenced field as `string` and not as `number`.

Unfortunately I don't know how to write the code to make a PR.

![image](https://user-images.githubusercontent.com/30607/209799912-71033b30-2e61-4c41-a6b3-76005c76e4fb.png)

### How to reproduce

The command

```
let df = open-df --infer-schema 200000 input.csv
```

The input file

```
comune,codice_regione,codice_provincia,codice_comune,denominazione_comune,sigla_provincia,data_entrata_in_carica
030151360,03,015,1360,POLPENAZZE DEL GARDA,BS,13/10/2021
030120530,03,012,0530,CAROBBIO DEGLI ANGELI,BG,04/10/2021
020040580,02,004,0580,SAINT-DENIS,AO,23/09/2020
030150870,03,015,0870,LOSINE,BS,04/10/2021
190480090,19,048,0090,CAPO D'ORLANDO,ME,27/10/2021
```

### Expected behavior

To have mapped correctly string fields as string, without truncating the original values

### Configuration

| key                | value                                           |
| ------------------ | ----------------------------------------------- |
| version            | 0.72.1                                          |
| branch             | main                                            |
| commit_hash        | 850ecf648a225eac2ae74846cec2ed47b3399a76        |
| build_os           | linux-x86_64                                    |
| build_target       | x86_64-unknown-linux-gnu                        |
| rust_version       | rustc 1.65.0 (897e37553 2022-11-02)             |
| rust_channel       | 1.65.0-x86_64-unknown-linux-gnu                 |
| cargo_version      | cargo 1.65.0 (4bc8f24d3 2022-10-20)             |
| pkg_version        | 0.72.1                                          |
| build_time         | 2022-12-03 12:31:31 +01:00                      |
| build_rust_channel | release                                         |
| features           | database, dataframe, default, trash, which, zip |
| installed_plugins  |                                                 |

**Comment by fdncred** [comment]
We do have something that may help in `from csv` but I'm not sure that will work for you, `--no-infer`.  This command will open it as a csv, not infer the data type, and convert it to a dataframe.

```
open -r input.csv | from csv --no-infer | into df
```
<img width="1533" alt="Screenshot 2022-12-28 at 8 04 07 AM" src="https://user-images.githubusercontent.com/343840/209823954-24513798-7cfc-4b64-bae9-942af1d5f319.png">

**Comment by aborruso** [comment]
> We do have something that may help in `from csv` but I'm not sure that will work for you, `--no-infer`. This command will open it as a csv, not infer the data type, and convert it to a dataframe.

Hi @fdncred I know it. 
My goal is not to solve my problem. Among other things, as I wrote, in this way everything becomes text, even those fields that are numbers; and so I would solve the case of text fields interpreted as strings, but I would have the problem of numeric fields becoming strings.

Since this great product does field inferencing, I want to point out that it doesn't do it well and that a new check rule could be added.
CSV files are still among the most popular data formats, and many times you have text fields, with only numbers inside.
If it were possible I wish it was added a subrule for all fields mapped as numbers: if there are cells that start with a zero, not followed by a `,` or `.`, set that inferenced field as string and not as number.

**Comment by fdncred** [comment]
Yes, I see what you mean. Unfortunately, it looks like the code that works on CSV files directly lies within the polars crate. The only solution I could find in dataframes is to allow you to pass in a schema. The polars crate supports that so then you could say column1 should be treated as a string and column 2 should be treated as an int, etc.

On the non-polars side (regular csv files), I think what you're suggesting would be possible because right now it's just calling this with `--no-infer`
```rust
            if no_infer {
                output_row.push(Value::String {
                    span,
                    val: value.into(),
                });
                continue;
            }
```
so that function `from_delimited_string_to_value()` could be made smarter as you suggest. I think we'd support a PR for smarter text processing.

**Comment by aborruso** [comment]
> so that function `from_delimited_string_to_value()` could be made smarter as you suggest. I think we'd support a PR for smarter text processing.

Thank you very much

**Comment by aborruso** [comment]
@fdncred I have add an interesting reply here
https://github.com/pola-rs/polars/issues/5924#issuecomment-1366886802

**Comment by fdncred** [comment]
ya, that is interesting. although, it looks like we'd have to vendor polars to make our own regex, which I'm sure it wouldn't happen. I still think the right way to go is just to expose the schema.
